package-lock.json

```diff
@@ -49,9 +49,10 @@
       "integrity": "sha512-W7tfG7vMOGtD30sHoZSSc/JVYiyDPEyQVso/Zz+/uQd0B0L46gtC+pHha5FFMRpil6fm/AoEcRWyOVi4+E/f8w=="
     },
     "tree-sitter-c": {
-      "version": "git+ssh://git@github.com/tree-sitter/tree-sitter-c.git#a2b7bac3b313efbaa683d9a276ff63cdc544d960",
```

Comment:
I actually have a git dependency to automatically be updated on every change of the C parser. There is workflow that automatically bumps to the newest commit on a cron-basis.

Is there actually somebody using tree-sitter-glsl from NPM? This would be the only situation where relying on a git-dependency would be a challenge. I assume most users will directly consume `parser.c` from this repo and since every commit in the upstream repo is potentially a breaking change semver doesn't rely make sense for tree-sitter parser (except upstream parser would really do a 0.minor version bump if they do a breaking change)

Before, what would be on NPM would be semi-random as not always new versions would be uploaded and I didn't want to be surprised by a huge amount of changes at once. I guess this has changed now since @amanq has taken over the upstream and pushes to NPM should now be automated.

I would prefer to stay with a git dependency. This way I can update all three parsers that depend on tree-sitter-c and tree-sitter-cpp when the change occurs. I can do releases with fixed dependency so downstream users can rely that released versions only rely on fixed versions. I'm currently thinking of a way of how a could get notified when there's a moment when I need to cut a release. I typically get notified on commits in tree-sitter-c, tree-sitter-cpp but an automation would be better.

Reply:
nvim-treesitter currently follows a rolling release model. When tree-sitter-c does a breaking change that affects the C queries all derived languages must follow immediately to be compatible with current Git tree-sitter-c. This was also a bit the motivation for the Git dependency. Although, tree-sitter-cpp uses a versioned dependency on tree-sitter-c which invalidates my arguments. At the moment, when tree-sitter-c does a breaking change and tree-sitter-cpp has not bumped it's tree-sitter-c dependency tree-sitter-hlsl and tree-sitter-cuda will fail on CI with their update workflow until upstream has settled again onto some consistent state.